src/beacons/use-should-show-share-travel-habits-screen.tsx

```diff
@@ -37,19 +41,12 @@ export const useShouldShowShareTravelHabitsScreen = (
   const shouldShowShareTravelHabitsScreen =
     enabled && !isConsentGranted && sessionCount > runAfterSessionsCount;
 
-  const updateCount = useCallback(
-    async (currentCount: number) => {
-      if (utilizeThisHookInstanceForSessionCounting) {
-        const newCount = currentCount + 1;
-        storage.set(shareTravelHabitsSessionCountKey, JSON.stringify(newCount));
-        sessionCountRef.current = newCount;
-        setSessionCount(newCount);
-      } else {
-        setSessionCount(currentCount);
-      }
-    },
-    [utilizeThisHookInstanceForSessionCounting],
-  );
+  const updateCount = useCallback(async (currentCount: number) => {
+    const newCount = currentCount + 1;
+    storage.set(shareTravelHabitsSessionCountKey, JSON.stringify(newCount));
```

Comment:
Is no matter of this PR, but I was wondering why is not possible to just send newCount directly and in the wrappers we could encode, decode the json as in the documentation: https://react-native-async-storage.github.io/async-storage/docs/usage

Reply:
I also think it should be fine to use JSON.stringify inside `storage.set`.
Loading values from storage however I think needs to know specifically whether it should be a string, int, json object etc